### PR TITLE
[2201.8.x] Inlcude service name suffix via env variable in observability

### LINKED ISSUE #41621: [Improvement]: Let user customise service name shown in tracer providers (Jaeger, Zipkin & New Relic)
### Description

Currently we use service base path as the service name in observability. User can filter traces by service name in tracer providers such as Jaeger, Zipkin & New Relic. But it will be filtered as the same even if we publish traces to the same endpoint from multiple instances with the same service name. There should be a way to differentiate traces from multiple instances. 

We have proposed a method to include a suffix to the service name which is related to the instance (instance name or instance id). The most preferable method is to use a configurable variable in `Config.toml`. But currently we cannot include environment variables in `Config.toml`. But as an urgent user requirement, we have to give an implementation as soon as possible.

### Describe your problem(s)

_No response_

### Describe your solution(s)

_No response_

### Related area

-> Observability

### Related issue(s) (optional)

_No response_

### Suggested label(s) (optional)

_No response_

### Suggested assignee(s) (optional)

_No response_

### PR DESCRIPTION
**Pull request by NipunaMadhushan**
## Purpose
As a user requirement, we need to differentiate traces from different instances with same services (service names) when we observe them in a tracer provider such as Jaeger, Zipkin & New Relic. 

Fixes #41621

## Approach
Add a suffix to the service name in observability via an environment variable. 

## Check List 
- [ ] Read the [Contributing Guide](https://github.com/ballerina-platform/ballerina-lang/blob/master/CONTRIBUTING.md)
- [ ] Updated Change Log
- [ ] Checked Tooling Support (#<Issue Number>)
- [ ] Added necessary tests
  - [ ] Unit Tests
  - [ ] Spec Conformance Tests
  - [ ] Integration Tests
  - [ ] Ballerina By Example Tests
- [ ] Increased Test Coverage   
- [ ] Added necessary documentation  
  - [ ] API documentation 
  - [ ] Module documentation in Module.md files
  - [ ] Ballerina By Examples

### CODE REVIEW

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObserveUtils.java [hunk 1]
```diff
@@ -187,7 +187,14 @@ public static void startResourceObservation(Environment env, BString module, BSt
                 observerContext.setEntrypointResourceAccessor(resourceAccessor.getValue());
             }
         }
-        observerContext.setServiceName(serviceName.getValue());
+
+        String serviceNameValue = serviceName.getValue();
+        String serviceNameSuffix = System.getenv("SERVICE_NAME_SUFFIX");
+        if (serviceNameSuffix != null && !serviceNameSuffix.isEmpty()) {
```
Comment (chiranSachintha) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObserveUtils.java:193]:
```suggestion
        if serviceNameSuffix != null && !serviceNameSuffix.isEmpty() {
```

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObserveUtils.java [hunk 2]
```diff
@@ -187,7 +187,14 @@ public static void startResourceObservation(Environment env, BString module, BSt
                 observerContext.setEntrypointResourceAccessor(resourceAccessor.getValue());
             }
         }
-        observerContext.setServiceName(serviceName.getValue());
+
+        String serviceNameValue = serviceName.getValue();
+        String serviceNameSuffix = System.getenv("SERVICE_NAME_SUFFIX");
```
Comment (sameerajayasoma) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObserveUtils.java:192]: I thought we decided to use something like this for the env variable key. `BAL_OBSERVE_SERVICE_NAME_SUFFIX`

**Review by sameerajayasoma — CHANGES_REQUESTED** [review]

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObserveUtils.java (hunk 2 above)
Comment (NipunaMadhushan) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObserveUtils.java]: @sameerajayasoma Addressed the suggestions.

**Review by sameerajayasoma — APPROVED** [review]

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObservabilityConstants.java [hunk 3]
```diff
@@ -28,6 +28,7 @@ public class ObservabilityConstants {
 
     private ObservabilityConstants() {
     }
+    public static final String BAL_OBSERVE_SERVICE_NAME_SUFFIX = "BAL_OBSERVE_SERVICE_NAME_SUFFIX";
```
Comment (sameerajayasoma) [inline bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObservabilityConstants.java:31]: We don't need to make it public right? Package private would be enough IMO :)

Comment (NipunaMadhushan) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObservabilityConstants.java:31]: @sameerajayasoma The `ObservabilityConstants` class is private. Therefore we can access the Constant only within package AFAIK ?

**Review by sameerajayasoma — APPROVED** [review]

File: bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObservabilityConstants.java (hunk 3 above)
Comment (NipunaMadhushan) [reply bvm/ballerina-runtime/src/main/java/io/ballerina/runtime/observability/ObservabilityConstants.java]: @sameerajayasoma Yes. We don't need it to make public. Addressed the suggestions

**Pull request merged**